File: backend-java/src/main/java/com/collaborative/dashboard/model/Task.java
package com.collaborative.dashboard.model;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "tasks")
public class Task {
    @Id
    private String id;
    private String title;
    private String description;
    private String boardId;
    private String columnId;
    private int position;
    private String assignee;
    private String priority;
    private Instant dueDate;
    private List<String> tags = new ArrayList<>();
    private String createdBy;
    private Instant createdAt;
    private Instant updatedAt;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getBoardId() { return boardId; }
    public void setBoardId(String boardId) { this.boardId = boardId; }
    public String getColumnId() { return columnId; }
    public void setColumnId(String columnId) { this.columnId = columnId; }
    public int getPosition() { return position; }
    public void setPosition(int position) { this.position = position; }
    public String getAssignee() { return assignee; }
    public void setAssignee(String assignee) { this.assignee = assignee; }
    public String getPriority() { return priority; }
    public void setPriority(String priority) { this.priority = priority; }
    public Instant getDueDate() { return dueDate; }
    public void setDueDate(Instant dueDate) { this.dueDate = dueDate; }
    public List<String> getTags() { return tags; }
    public void setTags(List<String> tags) { this.tags = tags; }
    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}

File: frontend/src/models/task.ts
import { IUserSummary } from './board';

export interface ITask {
    _id: string;
    title: string;
    description?: string;
    boardId: string;
    columnId: string;
    position: number;
    assignee?: IUserSummary;
    priority?: 'low' | 'medium' | 'high';
    dueDate?: string;
    tags?: string[];
    createdBy: IUserSummary;
    createdAt: string;
    updatedAt: string;
}

export interface ICreateTaskInput {
  title: string;
  description?: string;
  boardId: string;
  columnId: string;
  position: number;
  assignee?: string;
  priority?: 'low' | 'medium' | 'high';
  dueDate?: string;
  tags?: string[];
  createdBy: string;
}

export interface ITaskListResponse {
    success: boolean;
    tasks: ITask[];
}

export interface ITaskResponse {
    success: boolean;
    task: ITask;
}
